src/snap.c

```diff
@@ -23,19 +23,22 @@
 
 #include "utils.h"
 
-bool verify_appname(const char *appname)
+bool verify_executable_name(const char *executable_name)
 {
-	// snappy appname is of form:
-	// snap.<name>.<app>
+	// The executable name is of form:
+	// snap.<name>.(<appname>|hook.<hookname>)
 	// - <name> must start with lowercase letter, then may contain
 	//   lowercase alphanumerics and '-'
-	// - <app> may contain alphanumerics and '-'
-	const char *whitelist_re = "^snap\\.[a-z][a-z0-9-]*\\.[a-zA-Z0-9-]+$";
+	// - <appname> may contain alphanumerics and '-'
+	// - <hookname must start with a lowercase letter, then may
+	//   contain lowercase letters and '-'
+	const char *whitelist_re =
+	    "^snap\\.[a-z](-?[a-z0-9])*\\.([a-zA-Z0-9](-?[a-zA-Z0-9])*|hook\\.[a-z](-?[a-z])*)$";
```

Comment:
It's [here](https://github.com/snapcore/snapd/blob/master/snap/validate.go), are you saying you want that link in the code?